app/src/main/java/com/example/wnews/views/home/news/detail/DetailFragment.kt

```diff
@@ -1,30 +1,110 @@
+import com.google.android.material.snackbar.Snackbar
 
 
-class DetailFragment : Fragment(R.layout.fragment_detail) {
+class DetailFragment : Fragment(R.layout.fragment_detail), NewsView {
 
     private var binding: FragmentDetailBinding? = null
-    lateinit var newsObject: News
+    private lateinit var newView: View
+    private lateinit var userAuth: UserAuth
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
         binding = FragmentDetailBinding.bind(view)
 
+        newView = view
+
         val activity: DetailActivity? = activity as DetailActivity?
-        newsObject = activity!!.newsObject
 
-        setContent()
+        val intentNewsId = activity!!.newsId
+
+        val news =
+            DetailPresenter().arrayListNews.find { it.newsId == intentNewsId }
+
+        initializeProvider()
+        setContent(news)
+        setListener(news)
+    }
+
+    private fun initializeProvider() {
+        userAuth = UserProvider.userAuth
     }
 
-    private fun setContent() {
-        binding!!.textViewTitle.text = newsObject.title
-        binding!!.textViewDetail.text = newsObject.detail
+    private fun setContent(news: News?) {
+        binding!!.textViewTitle.text = news!!.title
+        binding!!.textViewDetail.text = news.detail
+        binding!!.bgImageUrl.setImageURI(news.imageUrl)
+
+        if (DetailPresenter().isUserLikeNews(news.like)) {
+            binding!!.imageButtonLikeDetail.setImageResource(R.drawable.ic_like_on)
+        }
     }
+
+    private fun setListener(news: News?) {
+
+        binding!!.imageButtonBackDetail.setOnClickListener {
+            val intent = Intent()
+            requireActivity().setResult(RESULT_OK, intent)
+            requireActivity().finish()
+        }
+
+        binding!!.imageButtonLikeDetail.setOnClickListener {
+
+            val newsToUpdateLike =
+                DetailPresenter().arrayListNews.find { it.newsId == news!!.newsId }
+
+
+            if (DetailPresenter().isUserLikeNews(newsToUpdateLike!!.like)) {
+
+                newsToUpdateLike.like = newsToUpdateLike.like.filter { it != userAuth.userAuthId }
+                binding!!.imageButtonLikeDetail.setImageResource(R.drawable.ic_like_off)
+
+            } else {
+                newsToUpdateLike.like = newsToUpdateLike.like.plusElement(userAuth.userAuthId)
+                binding!!.imageButtonLikeDetail.setImageResource(R.drawable.ic_like_on)
+            }
+
+            DetailPresenter().onResponseUpdateNewsLike(news!!.newsId, this)
+        }
+    }
+
+    override fun onSuccessResponse() {}
+
+    override fun onFailureResponse(message: String) {
+        val newMessage = if (message.isEmpty()) {
+            getString(R.string.server_error)
+        } else {
+            message
+        }
+
+        Snackbar.make(newView, newMessage, Snackbar.LENGTH_LONG).show()
+    }
+
+
+    /*NOT IMPLEMENTED*/
```

Comment:
Why do you need all of this methods in the interface if they are not gonna be used? Or is this intended to be used in the future?